### PR TITLE
Feat(Torii-core): Handle entity deletion events

### LINKED ISSUE #1349: [Torii-core] Handle entity deletion events properly
Currently, Torii is not deleting entities from internal db when a corresponding deletion event takes place onchain. 

- [ ] Implement processor to handle `StoreDelRecord` events that are generated from the world. 
https://github.com/dojoengine/dojo/tree/main/crates/torii/core/src/processors

- [ ] Delete db records from entities tables and all related model data from tables
https://github.com/dojoengine/dojo/blob/e599968f85f75e879346a789ae4d33ca8f590987/crates/torii/core/src/sql.rs#L152

### PR DESCRIPTION
**Pull request by gianalarcon**
- [x]  Implement processor to handle `StoreDelRecord` events that are generated from the world. 
- [x] Implement fn process() for `StoreDelRecordProcessor`
- [x] Implement fn build_delete_entity_queries_recursive() on fn delete_entity()
- [x] Improve current model_test
- [x] Implement delete tests 

Closes #1349

@broody

### CODE REVIEW

File: examples/spawn-and-move/src/actions.cairo [hunk 1]
```diff
@@ -129,5 +144,10 @@ mod tests {
         let new_position = get!(world, caller, Position);
         assert(new_position.vec.x == 11, 'position x is wrong');
         assert(new_position.vec.y == 10, 'position y is wrong');
+
+        actions_system.delete_move();
+        // check world state
+        let moves = get!(world, caller, Moves);
+        assert(moves.remaining == 0, 'moves is wrong');
```
Comment (broody) [inline examples/spawn-and-move/src/actions.cairo:151]: This test is incorrectly passing as remaining is already zero at this point. 

But for some reason the `delete!` macro seems to be failing (maybe that was ur earlier hunch!). I'll file an issue

Comment (gianalarcon) [reply examples/spawn-and-move/src/actions.cairo:151]: Oh. Gonna fix it! Thank you so much

File: crates/torii/types-test/src/contracts.cairo [hunk 2]
```diff
@@ -120,5 +120,10 @@ mod records {
             };
             return ();
         }
-    }
-}
+        // Implemment fn delete, input param: record_id
+        fn delete(self: @ContractState, record_id: u32) {
+            let world = self.world_dispatcher.read();
+            let (record, record_sibling) = get!(world, record_id, (Record, RecordSibling));
```
Comment (broody) [inline crates/torii/types-test/src/contracts.cairo:126]: Let's also get and delete `Subrecord`. The subrecord_id is just record_id + 1

**Review by broody — COMMENTED** [review]
Recursive model deletion looking good, just some minor comments on testing

File: crates/torii/core/src/processors/store_del_record.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        event: &Event,
+    ) -> Result<(), Error> {
+        let name = parse_cairo_short_string(&event.data[MODEL_INDEX])?;
+        info!("store delete record: {}", name);
+
+        let model = db.model(&name).await?;
+
+        let keys_start = NUM_KEYS_INDEX + 1;
+        let keys = event.data[keys_start..].to_vec();
+        info!("keys: {:?}", keys);
```
Comment (broody) [inline crates/torii/core/src/processors/store_del_record.rs:54]: Remove debugging info

**Review by broody — COMMENTED** [review]
Just one more debug print remove and it LGTM

**Review by broody — APPROVED** [review]

**Pull request merged**